main.py

```
@@ -0,0 +1,1832 @@
+from datetime import datetime, timedelta
+
+# GUI Framework
+import customtkinter as ctk
+import tkinter as tk
+from tkinter import messagebox, filedialog
+
+# 3D Graphics & Animations
+import pygame
+import moderngl
+import numpy as np
+from OpenGL.GL import *
```

Ruff (0.12.2)

`from OpenGL.GL import *` used; unable to detect undefined names

(F403)